engine/src/flutter/lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -542,10 +501,19 @@ void ImageDecoderImpeller::Decode(fml::RefPtr<ImageDescriptor> descriptor,
         auto max_size_supported =
             context->GetResourceAllocator()->GetMaxTextureSizeSupported();
 
+        bool is_gpu_disabled = false;
+        gpu_disabled_switch->Execute(
+            fml::SyncSwitch::Handlers()
+                .SetIfFalse([&] { is_gpu_disabled = false; })
+                .SetIfTrue([&] { is_gpu_disabled = true; }));
+
         // Always decompress on the concurrent runner.
-        auto bitmap_result = DecompressTexture(
-            raw_descriptor, target_size, max_size_supported,
-            /*supports_wide_gamut=*/supports_wide_gamut,
+        DecompressResult bitmap_result = DecompressTexture(
+            raw_descriptor,                               //
+            target_size,                                  //
+            max_size_supported,                           //
+            /*supports_wide_gamut=*/supports_wide_gamut,  //
+            /*is_gpu_disabled=*/is_gpu_disabled,          //
```

Comment:
This has to be called inside the sync switch handler otherwise you can lose gpu access while uploading.

Reply:
hmm. Let me refactor this. WIP

Reply:
ahh damn, so this is actually pretty tricky. 

The user may have requested a resize. We want to use a GPU resize, so we need to check if the GPU is enabled to fall back to CPU resize.

But we could actually lose GPU access between the decode task (which runs on the iO thread) and the upload task (which runs on the worker thread). That means we could prepare the image for GPU upload then fail and end up not resizing it correctly.

I don't think there is a way to fix this.